src/net_processing.cpp

```diff
@@ -2992,17 +2992,15 @@ void ProcessMessage(
                 uint32_t nFetchFlags = GetFetchFlags(pfrom);
                 const auto current_time = GetTime<std::chrono::microseconds>();
 
-                if (!State(pfrom.GetId())->m_wtxid_relay) {
-                    for (const CTxIn& txin : tx.vin) {
-                        // Here, we only have the txid (and not wtxid) of the
-                        // inputs, so we only request parents from
-                        // non-wtxid-relay peers.
-                        // Eventually we should replace this with an improved
-                        // protocol for getting all unconfirmed parents.
-                        CInv _inv(MSG_TX | nFetchFlags, txin.prevout.hash);
-                        pfrom.AddKnownTx(txin.prevout.hash);
-                        if (!AlreadyHave(_inv, mempool)) RequestTx(State(pfrom.GetId()), ToGenTxid(_inv), current_time);
-                    }
+                for (const CTxIn& txin : tx.vin) {
+                    // Here, we only have the txid (and not wtxid) of the
+                    // inputs, so we only request in txid mode, even for
+                    // wtxidrelay peers.
+                    // Eventually we should replace this with an improved
+                    // protocol for getting all unconfirmed parents.
+                    CInv _inv(MSG_TX | nFetchFlags, txin.prevout.hash);
+                    pfrom.AddKnownTx(txin.prevout.hash);
+                    if (!AlreadyHave(_inv, mempool)) RequestTx(State(pfrom.GetId()), ToGenTxid(_inv), current_time);
```

Comment:
I believe this wouldn't work at the moment, because there's a check:
```
            if (State(pfrom.GetId())->m_wtxid_relay) {
                if (inv.type == MSG_TX) continue;
            } else {
                if (inv.type == MSG_WTX) continue;
            }
```

Does it rely on some other pending PR? Is that what you referring to: "Based on a commit by Anthony Towns."?

Reply:
Or maybe it's just missing a commit to remove this check, I don't know :)

Reply:
This is in INV handling. When requesting parents of an orphan, there is no INV involved: a GETDATA is sent directly in response to receiving the TX.

Am I missing something why changing this is needed?

Reply:
This works as-is for me both live against a mainnet wtxid relay peer, and via the modified test.

Reply:
You are right, I confused INV/GETDATA flow. Feel free to mark as resolved :)